File: app/src/main/java/com/tutv/android/di/ContainerLocator.java
package com.tutv.android.di;

import android.content.Context;

public class ContainerLocator {
    private static Container container;

    private ContainerLocator() {}

    public static Container locateComponent(final Context context) {
        if(container == null)
            setComponent(new AppContainer(context));

        return container;
    }

    public static void setComponent(final Container container) {
        ContainerLocator.container = container;
    }
}

File: app/src/main/java/com/tutv/android/di/ContainerLocator.kt
package com.tutv.android.di

import android.content.Context

// TODO: refactor the @JvmStatic method on UI migration!
object ContainerLocator {
    private lateinit var container: Container

    @JvmStatic
    fun locateContainer(context: Context): Container {
        if (!this::container.isInitialized) setContainer(AppContainer(context))
        return container
    }

    private fun setContainer(container: Container) {
        ContainerLocator.container = container
    }
}